# 10 Questions

The purpose of this assignment is to pick 10 questions from [this document](https://docs.google.com/document/d/1l8ubmYkEA0wXas9S6HOb6o2lPCPkjgSPL0R5O1v18Bw/edit) and answer them here.  The questions are also listed below.

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
np.random.seed(1547)
df = pd.read_csv('datasets/titanic.csv')

### 1.) Given a passenger had no siblings/spouse/children onboard, what are the odds of their survival?

In [22]:
df[df['SibSp'] == 0].groupby(['Survived']).size().div(len(df))

Survived
0    0.446689
1    0.235690
dtype: float64

#### ANSWER: 23.6%

### 2.) Given that a passenger is from Southampton, what are the chances that they survived?

In [24]:
df[df['Embarked'] == 'S'].groupby(['Survived']).size().div(len(df))

Survived
0    0.479237
1    0.243547
dtype: float64

#### ANSWER: 24.4%

### 3.) Given that a passenger survived, what are the chances they were 2nd class?

In [27]:
df[df['Survived'] == 1].groupby(['Pclass']).size().div(len(df))

Pclass
1    0.152637
2    0.097643
3    0.133558
dtype: float64

#### ANSWER: 9.8%

### 4.) Given that a passenger survived, what are the odds their age is under 18?

In [29]:
df[df['Survived'] == 1].groupby(df['Age'] < 18).size().div(len(df))

Age
False    0.315376
True     0.068462
dtype: float64

#### ANSWER: 6.8%

### 5.) Given that a passenger was over 40, what are the odds he was middle class?

In [30]:
df[df['Age'] > 40].groupby(df['Pclass'] == 2).size().div(len(df))

Pclass
False    0.130191
True     0.038159
dtype: float64

#### ANSWER: 3.8%

### 6.) Given that a passenger survived, what are the odds that they are not first class and ander the age of 5?

In [31]:
df[df['Survived'] == 1].groupby((df['Pclass'] != 1) & (df['Age'] < 5)).size().div(len(df))

False    0.355780
True     0.028058
dtype: float64

#### ANSWER: 2.8%

### 7.) Given that a passenger is over 30 years old, what are the odds they paid over $15 for a ticket?

In [32]:
df[df['Age'] > 30].groupby(df['Fare'] > 15).size().div(len(df))

Fare
False    0.136925
True     0.205387
dtype: float64

#### ANSWER: 20.5%

### 8.) Given that a passenger was a female, what are the odds they had more than one sibling?

In [33]:
df[df['Sex'] == 'female'].groupby(df['SibSp'] > 1).size().div(len(df))

SibSp
False    0.314254
True     0.038159
dtype: float64

#### ANSWER: 3.8%

### 9.) Given that  a passenger is in 1st class, what are the chances that they paid less than $50 for their ticket

In [34]:
df[df['Pclass'] == 1].groupby(df['Fare'] < 50).size().div(len(df))

Fare
False    0.157127
True     0.085297
dtype: float64

#### ANSWER: 8.5%

### 10.) Given that a passenger is male, what are the odds they paid more than the average price for their ticket?

In [35]:
df[df['Sex'] == 'male'].groupby(df['Fare'] > df['Fare'].mean()).size().div(len(df))

Fare
False    0.527497
True     0.120090
dtype: float64

#### ANSWER: 12%